tests/compilers/compilation.rs

```diff
@@ -0,0 +1,207 @@
+use std::fs;
+use std::sync::Arc;
+use wasmer::*;
+use wasmer_engine::Engine;
+use wasmer_engine_universal::Universal;
+use wasmer_types::InstanceConfig;
+use wasmer_types::Type::I64;
+
+fn slow_to_compile_contract(n_fns: usize, n_locals: usize) -> Vec<u8> {
+    let fns = format!("(func (local {}))\n", "i32 ".repeat(n_locals)).repeat(n_fns);
+    let wat = format!(r#"(module {} (func (export "main")))"#, fns);
+    wat2wasm(wat.as_bytes()).unwrap().to_vec()
+}
+
+fn compile_uncached<'a>(
+    store: &'a Store,
+    engine: &'a dyn Engine,
+    code: &'a [u8],
+) -> Result<Arc<dyn wasmer_engine::Artifact>, CompileError> {
+    use std::time::Instant;
+    let now = Instant::now();
+    engine.validate(code)?;
+    let validate = now.elapsed().as_millis();
+    let now = Instant::now();
+    let res = engine.compile(code, store.tunables());
+    let compile = now.elapsed().as_millis();
+    println!("validate {} compile {}", validate, compile);
+    res
+}
+
+#[test]
+#[ignore]
+fn compilation_test() {
+    let compiler = Singlepass::default();
+    let engine = Universal::new(compiler).engine();
+    let store = Store::new(&engine);
+    for factor in 1..1000 {
+        let code = slow_to_compile_contract(3, 25 * factor);
+        match compile_uncached(&store, &engine, &code) {
+            Ok(art) => {
+                let serialized = art.serialize().unwrap();
+                println!(
+                    "{}: artefact is compiled, size is {}",
+                    factor,
+                    serialized.len()
+                );
+            }
+            Err(err) => {
+                println!("err is {:?}", err);
+            }
+        }
+    }
+}
+
+#[test]
+#[ignore]
+fn aurora_profiling() {
+    let contract_wasm = "/Volumes/Samsung_T5/near/aurora-engine/mainnet-release.wasm";
+    let compiler = Singlepass::default();
+    let engine = Universal::new(compiler).engine();
+    let store = Store::new(&engine);
+    let code = fs::read(contract_wasm).unwrap();
+    match compile_uncached(&store, &engine, &code) {
+        Ok(art) => unsafe {
+            let serialized = art.serialize().unwrap();
+            println!("artefact is compiled, size is {}", serialized.len());
+            let module = wasmer::Module::deserialize(&store, serialized.as_slice()).unwrap();
+            let instance = Instance::new_with_config(
+                &module,
+                InstanceConfig::default().with_profile(),
+                &imports! {
```

Comment:
I think this test is better suited for nearcore or even aurora's engine-tests, rather than here. Having to replicate all the `VMLogic` exports here seems iffy.

Reply:
Will rework this test by checking symbol map.